modules/tensor_mechanics/src/materials/HillConstants.C

```diff
@@ -184,47 +184,19 @@ HillConstants::rotateHillConstants(const std::vector<Real> & hill_constants_inpu
   // Transformed the Hill tensor given the total rotation matrix
   // https://scicomp.stackexchange.com/questions/35600/4th-order-tensor-rotation-sources-to-refer
   DenseMatrix<Real> transformation_matrix_n(6, 6);
-  transformation_matrix_n(0, 0) = trm(0, 0) * trm(0, 0);
-  transformation_matrix_n(0, 1) = trm(0, 1) * trm(0, 1);
-  transformation_matrix_n(0, 2) = trm(0, 2) * trm(0, 2);
-  transformation_matrix_n(0, 3) = trm(0, 1) * trm(0, 2);
-  transformation_matrix_n(0, 4) = trm(0, 0) * trm(0, 2);
-  transformation_matrix_n(0, 5) = trm(0, 0) * trm(0, 1);
-
-  transformation_matrix_n(1, 0) = trm(1, 0) * trm(1, 0);
-  transformation_matrix_n(1, 1) = trm(1, 1) * trm(1, 1);
-  transformation_matrix_n(1, 2) = trm(1, 2) * trm(1, 2);
-  transformation_matrix_n(1, 3) = trm(1, 1) * trm(1, 2);
-  transformation_matrix_n(1, 4) = trm(1, 0) * trm(1, 2);
-  transformation_matrix_n(1, 5) = trm(1, 0) * trm(1, 1);
-
-  transformation_matrix_n(2, 0) = trm(2, 0) * trm(2, 0);
-  transformation_matrix_n(2, 1) = trm(2, 1) * trm(2, 1);
-  transformation_matrix_n(2, 2) = trm(2, 2) * trm(2, 2);
-  transformation_matrix_n(2, 3) = trm(2, 1) * trm(2, 2);
-  transformation_matrix_n(2, 4) = trm(2, 0) * trm(2, 2);
-  transformation_matrix_n(2, 5) = trm(2, 0) * trm(2, 1);
-
-  transformation_matrix_n(3, 0) = 2.0 * trm(1, 0) * trm(2, 0);
-  transformation_matrix_n(3, 1) = 2.0 * trm(1, 1) * trm(2, 1);
-  transformation_matrix_n(3, 2) = 2.0 * trm(1, 2) * trm(2, 2);
-  transformation_matrix_n(3, 3) = trm(1, 1) * trm(2, 2) + trm(1, 2) * trm(2, 1);
-  transformation_matrix_n(3, 4) = trm(1, 0) * trm(2, 2) + trm(1, 2) * trm(2, 0);
-  transformation_matrix_n(3, 5) = trm(1, 0) * trm(2, 1) + trm(1, 1) * trm(2, 0);
-
-  transformation_matrix_n(4, 0) = 2.0 * trm(0, 0) * trm(2, 0);
-  transformation_matrix_n(4, 1) = 2.0 * trm(0, 1) * trm(2, 1);
-  transformation_matrix_n(4, 2) = 2.0 * trm(0, 2) * trm(2, 2);
-  transformation_matrix_n(4, 3) = trm(0, 1) * trm(2, 2) + trm(0, 2) * trm(2, 1);
-  transformation_matrix_n(4, 4) = trm(0, 0) * trm(2, 2) + trm(0, 2) * trm(2, 0);
-  transformation_matrix_n(4, 5) = trm(0, 0) * trm(2, 1) + trm(0, 1) * trm(2, 0);
-
-  transformation_matrix_n(5, 0) = 2.0 * trm(0, 0) * trm(1, 0);
-  transformation_matrix_n(5, 1) = 2.0 * trm(0, 1) * trm(1, 1);
-  transformation_matrix_n(5, 2) = 2.0 * trm(0, 2) * trm(1, 2);
-  transformation_matrix_n(5, 3) = trm(0, 1) * trm(1, 2) + trm(0, 2) * trm(1, 1);
-  transformation_matrix_n(5, 4) = trm(0, 0) * trm(1, 2) + trm(0, 2) * trm(1, 0);
-  transformation_matrix_n(5, 5) = trm(0, 0) * trm(1, 1) + trm(0, 1) * trm(1, 0);
+
+  // construct "bond-like" rotation matrix
```

Comment:
Where is the bond-like transformation matrix documented?

Reply:
I cannot find where I read "bond-like" 🔫 , [this discussion](https://en.wikiversity.org/wiki/Talk:Coordinate_transformations) calls it _bound matrix_, but Google did not give me any useful definition of that either. I'll just remove the comment. 

Either way, the loop sets up exactly the same matrix that the unrolled version set up.

Reply:
Oh, I found it, "bond-like" is what it is called in https://scicomp.stackexchange.com/questions/35600/4th-order-tensor-rotation-sources-to-refer which is the link you added in the comment above...

Reply:
So, basically you did the math to condense the number of lines needed to compute the transformation matrix for compliance tensors, right?

Reply:
yes, that's all there is to this